lib/decompress.mli

```diff
@@ -237,6 +237,26 @@ type error_z_deflate = RFC1951 of RFC1951_deflate.error
 
 module Zlib_deflate : DEFLATE with type error = error_z_deflate
 
+type error_g_deflate = RFC1951 of RFC1951_deflate.error
+
+(* module Gzip_deflate : DEFLATE with type error = error_g_deflate *)
+module Gzip_deflate : sig
+  include DEFLATE with type error = error_g_deflate
+
+  val default :
+       witness:'a B.t
+    -> ?text:bool
+    -> ?header_crc:bool
+    -> ?extra:string option
+    -> ?name:string option
+    -> ?comment:string option
+    -> ?mtime:int
+    -> ?os:int
```

Comment:
RFC 1952 describes explicitly which values you can put in this field:

```
                 0 - FAT filesystem (MS-DOS, OS/2, NT/Win32)
                 1 - Amiga
                 2 - VMS (or OpenVMS)
                 3 - Unix
                 4 - VM/CMS
                 5 - Atari TOS
                 6 - HPFS filesystem (OS/2, NT)
                 7 - Macintosh
                 8 - Z-System
                 9 - CP/M
                10 - TOPS-20
                11 - NTFS filesystem (NT)
                12 - QDOS
                13 - Acorn RISCOS
               255 - unknown
``` 

It should be interesting to replace `int` by an ADT and then, have two functions like:

```ocaml
type os

val os_of_int : int -> os
val int_of_os : os -> int
```

Reply:
likely rather :)
```OCaml
val os_of_int : int -> os option
```